style/glossary.md

```diff
@@ -0,0 +1,52 @@
+## What is this?
+
+This file is targeted at the translators and serves as a basic tutorial for translating this project. It includes two lists of terms:
+  - terms which do not get translated at all
+  - terminology translations (glossary)
+
+### What does not get translated?
+
+Class names and proper nouns do not get translated.
+
+
+### Table of Contents
+
+- [Glossary - Russian](#glossary-russian)
+  - [Elixir-related terms](#russian-elixir-related-terms)
+  - [CS terms](#russian-cs-terms)
+
+### <a name="glossary-russian"></a> Glossary - Russian
+
+#### <a name="russian-elixir-related-terms"></a> Elixir-related terms
+
+- alias - псевдоним
+- guard - ограничитель
```

Comment:
(Somewhat) Commonly used Russian translation for guard is "страж" or "охранное выражение" rather than "ограничитель".

Reply:
Great, thank you :)

Reply:
If I am not mistaken, the RU glossary lists `guard` with:

```
guard (in a pattern match) | предохранитель [Виль15]; охранное выражение
```

Reply:
Alright, after talking with @Bodigrim, we decided that we will stick with страж, as it will give the text some fluency.

Reply:
Страж is a good one. As for "предохранитель [Виль15]" I know the source and I'm pretty sure no one in practice uses this translation no matter that there was one book that used it. That wiki page with Russian translations holds a number of such practically unused translations unfortunately. But when you try to argue about that they point you to that stupid book [Виль15] 😔

Reply:
Yeah as someone who had to review some French translations of English terms, some authors desperately want to use their weird terminology. :)
@ulysses4ever If you want, we can pioneer new translations for these terms?

Reply:
:-) I'd rather not! Страж is good enough. For the rest: it's mostly fine. I think that translations from [Виль15] are significantly off in a number of cases and better be omitted in those cases. Below are some examples but really I don't want to hold off this PR: the translation debate should probably happen somewhere else...

Here are examples where it's better to leave just one version (and many times it's not [Виль15], although there' are exceptions).

```
| ad hoc (polymorphism) | специализированный [Брон14] |
| concurrency | конкурентность [Браг13М] |
| identity function, morphism | тождественное преобразование |
| instance | экземпляр [Виль15] |
| monad transformer | монадный трансформер |
| parallelism | распараллеливание [Браг13М]; параллелизм [Виль15] |
| section (частичное применение оператора) | сечение |
| smart constructor | умный конструктор |
| strict/eager (evaluation) | строгое; энергичное [Зеф09] (вычисление) |
```

There are cases where there's no alternative but there are in fact more wide-spread versions, e.g.:

```
| bind | (монадическое) связывание |
| default definition | определение по умолчанию |
| derive (an instance) | вывести (экземпляр) |
```

One important case where there's no alternative is "list comprehensions and specifiers" (what are specifiers??). I don't think there's a wide-spread enough translation for "list comprehension", but I'm positive that "формитрователь" is not a proper Russian word, and it sounds very weird. I personally translate "list comprehension" as "генератор списка", literally "list generator". It could clash with any other notion of "generator" but luckily in Haskell we don't really have anything like that. (For Python it would cause more trouble.)